### PR TITLE
Add command to call tsp compilation for test projects

### PR DESCRIPTION
**Pull request by pshao25**
This is part of https://github.com/Azure/autorest/issues/4947

We could call `npm run generate swagger {project folder}` to run the compilation.

### CODE REVIEW

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts [hunk 1]
```diff
@@ -67,6 +93,9 @@ async function main() {
     const folder = folders[i];
     try {
       await generateTypespec(repoRoot, folder, debug, i % 2 === 0);
+      if (swagger) {
```
Comment (timotheeguerin) [inline packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:96]: Is that not going to crash if they don’t have installed typespec and the dependencies yet?

Comment (pshao25) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:96]: I don't expect users to call it. But if you want, I could move the dev dependency to dependency. Is that what you are saying?

Comment (timotheeguerin) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:96]: Oh no is that just meant as test in this repo?

Comment (pshao25) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:96]: It's just test purpose.

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts [hunk 2]
```diff
@@ -24,6 +37,18 @@ export async function generateTypespec(repoRoot: string, folder: string, debug =
...
+export async function generateSwagger(folder: string) {
+  if (brownFieldProjects.includes(folder)) {
+    console.log(`Skipping brown field project ${folder}`);
+    return;
+  }
+
+  const { path: root } = await resolveProject(__dirname);
+  const path = join(root, "test", folder, "tsp-output");
+  const command = "tsp compile . --emit=@azure-tools/typespec-autorest";
+  execSync(command, { cwd: path, stdio: "inherit" });
```
Comment (live1206) [inline packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:49]: once we generated the swagger in the test project folder, will we keep them in the test project folder as well?

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts [hunk 3]
```diff
@@ -24,6 +37,18 @@ export async function generateTypespec(repoRoot: string, folder: string, debug =
   generate(repoRoot, swaggerPath, debug, isFullCompatible);
 }
 
+export async function generateSwagger(folder: string) {
+  if (brownFieldProjects.includes(folder)) {
```
[the remark this answers is not in the document]
Comment (pshao25) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:41]: My thought is we first enable green field and make the E2E comparison (compare the generated swagger and original swagger) work, then we could enable the brown field.

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts (hunk 2 above)
Comment (pshao25) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:49]: Yes!

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (live1206) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:41]: OK, if so please add a TODO to remove this skipping, the brown-field test project should compile eventually.

**Review by live1206 — APPROVED** [review]

File: packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (pshao25) [reply packages/extensions/openapi-to-typespec/test/utils/generate-typespec.ts:41]: Oh right, I think I should remove this check.

**Pull request merged**